app/src/pages/clinic.tsx

```diff
@@ -216,12 +217,28 @@ export const getServerSideProps: GetServerSideProps = async ({
   locale,
   req,
 }) => {
-  return {
-    props: {
-      previousRoute: req.headers.referer,
-      ...(await serverSideTranslations(locale || 'en', ['common'])),
-    },
+  const prevRouteIndex = req.headers.referer?.lastIndexOf('/')
+  const previousRoute =
+    prevRouteIndex && req.headers.referer?.substring(prevRouteIndex)
+  let returnval: GetServerSidePropsResult<{ [key: string]: object | string }> =
+    {
+      props: {
+        previousRoute: previousRoute as string,
+        ...(await serverSideTranslations(locale || 'en', ['common'])),
+      },
+    }
+
+  if (!['/income', '/review', '/contact'].includes(previousRoute as string)) {
```

Comment:
It's possible for a user to go directly from the eligibility page to the clinic selection if they have chosen an approved adjunctive eligibility criteria.

```suggestion
  if (!['/income', '/review', '/contact', '/eligibility'].includes(previousRoute as string)) {
```